File: Data Structures & Algorithms/string-encode-and-decode/submission-2.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        output = ""
        for st in strs:
            output += str(len(st)) + "Ð" + st
        return output

    def decode(self, s: str) -> List[str]:
        output = []
        i = 0
        while i < len(s):
            j = i
            while s[j] != "Ð":
                j += 1
            st_len = int(s[i:j])
            word = s[j+1:j+1+st_len]
            output.append(word)
            i = j + 1 + st_len
        return output

File: Data Structures & Algorithms/longest-common-prefix/submission-2.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() { return String::new(); }

        for (i, ch) in strs[0].chars().enumerate(){
            for word in &strs[1..]{
                if word.chars().nth(i) != Some(ch) {
                    return strs[0][..i].to_string();
                }
            }

        }
        strs[0].clone()
    }
}
